### DISCUSSION TITLE
Feature request: recognise custom Pandas accessors in autocomplete, hover information, etc

### DISCUSSION THREAD

**Discussion by wurli** [posit-dev/positron#5067] [Feedback]
Pandas has an extension system which allows users to create [custom accessors](https://pandas.pydata.org/pandas-docs/stable/development/extending.html#registering-custom-accessors) for DataFrame objects. For example, one could define `DataFrame.utils.semi_join()` like so:

``` python

import pandas as pd

@pd.api.extensions.register_dataframe_accessor("utils")
class UtilsAccessor:
    def __init__(self, pandas_df):
        self._df = pandas_df
        
    def semi_join(self, right: pd.DataFrame, on=None, left_on=None, right_on=None):
        """Get all rows from ``left`` with a match in ``right``
        
        A simple helper to perform a filtering join without unnecessarily adding
        new columns to the DataFrame.

        :param right: Passed to ``pd.DataFrame.merge``
        :param on: Passed to ``pd.DataFrame.merge``
        :param left_on: Passed to ``pd.DataFrame.merge``
        :param right_on: Passed to ``pd.DataFrame.merge``
        
        :param returns: ``pd.DataFrame``
        """
        df = self._df

        if on is not None:
            right = right[on]
        if right_on is not None:
            right = right[right_on]

        return df.merge(right, on=on, left_on=left_on, right_on=right_on, how="inner")
```

Unfortunately, IDE features like autocomplete, type hits, hover documentation, etc, don't pick up accessors added in this way. NB, it [doesn't seem](https://stackoverflow.com/questions/66501738/pandas-dataframe-accessor-type-hints) like other IDEs support this feature either. It would be great if this could be added to Positron 🙂

Many thanks 🙏

**Comment by wesm** [comment]
In theory, it seems like it may be possible to hook this kind of thing into the jedi lsp/autocomplete infrastructure that powers the Positron Python experience. Could you open a feature request and it can be investigated in the future? Thanks for the suggestion

**Reply by wurli** [reply]
Thanks! #5077 opened 🙂
